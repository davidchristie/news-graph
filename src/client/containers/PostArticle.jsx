import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import { postArticle } from '../actions/articles'

export class PostArticle extends React.Component {
  constructor (props) {
    super(props)
    this.state = {value: ''}
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange (event) {
    this.setState({value: event.target.value})
  }
  handleSubmit (event) {
    event.preventDefault()
    const token = this.props.token
    const url = this.state.value
    this.props.dispatch(postArticle({token, url}))
    this.setState({value: ''})
  }
  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <div className='form-inline'>
          <input className='form-control' type='url' value={this.state.value} onChange={this.handleChange} placeholder='Article' />
          <input className='btn btn-outline-success' type='submit' value='POST' />
        </div>
      </form>
    )
  }
}

PostArticle.propTypes = {
  dispatch: PropTypes.func.isRequired,
  token: PropTypes.string.isRequired
}

const mapStateToProps = state => {
  const profile = state.app.account.profile
  return {
    token: profile ? profile.token : null
  }
}

export default connect(mapStateToProps)(PostArticle)
