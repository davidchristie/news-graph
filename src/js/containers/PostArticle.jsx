import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import { postArticle } from '../actions'

class PostArticle extends React.Component {
  constructor(props) {
    super(props)
    this.state = {value: ''}
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(event) {
    this.setState({value: event.target.value})
  }
  handleSubmit(event) {
    event.preventDefault()
    this.props.dispatch(postArticle(this.state.value))
    this.setState({value: ''})
  }
  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label>
          Article:
          <input type="url" value={this.state.value} onChange={this.handleChange} />
        </label>
        <input type="submit" value="POST" />
      </form>
    )
  }
}

PostArticle.propTypes = {
  dispatch: PropTypes.func.isRequired
}

export default connect()(PostArticle)
