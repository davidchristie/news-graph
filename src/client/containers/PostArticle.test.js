/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { PostArticle } from './PostArticle'

test('PostArticle renders correctly', () => {
  const dispatch = () => {}
  const wrapper = shallow(<PostArticle dispatch={dispatch} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
