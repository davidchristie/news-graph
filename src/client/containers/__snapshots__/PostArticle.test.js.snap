exports[`test PostArticle renders correctly 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="form-inline">
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Article"
      type="url"
      value="" />
    <input
      className="btn btn-outline-success"
      type="submit"
      value="POST" />
  </div>
</form>
`;
